Core/include/Acts/Utilities/Result.hpp

```diff
@@ -39,27 +39,24 @@ class Result {
   /**
    * Copy construction is disallowed
    */
-  Result(const Result<T, E>& other) = delete;
+  Result(const Result<T, E>&) = delete;
 
   /**
    * Assignment is disallowed
    */
-  Result<T, E>& operator=(const Result<T, E>& other) = delete;
+  Result<T, E>& operator=(const Result<T, E>&) = delete;
 
   /**
    * Move construction is allowed
    */
-  Result(Result<T, E>&& other) : m_var(std::move(other.m_var)){};
+  Result(Result<T, E>&&) = default;
```

Comment:
So, I [**explicitly** changed this from `default` to the explicit implementation](https://github.com/acts-project/acts/commit/e3695e6814ce8163da32e2bd67c4b6f8da3bf2eb#diff-e5d011aba905cf40f87d2fab575d1f65L52-R52). I don't quite remember the reason here anymore however. I vaguely remember this had to do with one of the clang builds.

Reply:
Same for the move assignment, btw.

Reply:
This looks exactly like the use case for defaulted constructors. There should be nothing special going on here. It might be that previous issues are related to the masked compiler warnings/errors that appeared with #267. Let me try to test this on top of #267 again.